tests/run/loops.scala

```diff
@@ -0,0 +1,37 @@
+import scala.util.{boundary, break}
+
+object loop:
```

Comment:
This is an interesting test, but it's not as good as it can be from a code generation point of view. In fact, the traditional continue+break loop features have a better implementation with *2* nested boundaries.
```scala
object loop:
  opaque type ExitLabel = Label[Unit]
  opaque type ContinueLabel = Label[Unit]

  inline def apply(inline op: (ExitLabel, ContinueLabel) ?=> Unit): Unit =
    boundary { exitLabel ?=>
      while true do
        boundary { continueLabel ?=>
          op(using exitLabel, continueLabel)
        }
    }
  end apply

  inline def exit()(using ExitLabel): Unit =
    break()

  inline def continue()(using ContinueLabel): Unit =
    break()
end loop
```
It might even make a good test case for the `dropBreaks` transformation: it should be able to eliminate *both* of those boundaries into `Labeled` blocks without `try`.

Reply:
Neat!

Reply:
I see two limitations we need to solve to make this work out of the box.

* Improve beta reduction. When we inline the `op` we generate a `$asInstanceOf` to refine the opaque types. This `asInstanceOf` is not currently handled by beta reduction.
* `op` arguments are not inlined. We create a proxy for these variables that we should elide. In theory, after beta reduction, these should be term references to the labels. But again, given that the Label type is opaque we insert an `asInstanceOf` that hides this detail when creating the beta reduction bindings. Alternatively, we could support inline lambdas with inline parameters and write `def apply(inline op: (inline ExitLabel, inline ContinueLabel) ?=> Unit)`.

Reply:
The type inference issues have been solved by making Label contravariant. I am including a test for the new kind of loop with a note that we cannot optimize it yet.